From 963f1be32b03e2437b8ff3032fc91d9887389035 Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Mon, 27 Jan 2020 11:14:32 +0300
Subject: [PATCH 1/1] Open keyboard FD in non-blocking mode otherwise obmc-ikvm
 will hang on try to write to keyboard if something is wrong on the USB side

---
 ikvm_input.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ikvm_input.cpp b/ikvm_input.cpp
index df12f27..e701025 100644
--- a/ikvm_input.cpp
+++ b/ikvm_input.cpp
@@ -29,7 +29,7 @@ Input::Input(const std::string& kbdPath, const std::string& ptrPath) :
 {
     if (!keyboardPath.empty())
     {
-        keyboardFd = open(keyboardPath.c_str(), O_RDWR | O_CLOEXEC);
+        keyboardFd = open(keyboardPath.c_str(), O_RDWR | O_CLOEXEC | O_NONBLOCK);
         if (keyboardFd < 0)
         {
             log<level::ERR>("Failed to open input device",
-- 
1.9.1

